pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java

```diff
@@ -168,13 +209,18 @@ public AbstractBuilder(TextIndexConfig other) {
       _luceneUseCompoundFile = other._luceneUseCompoundFile;
       _luceneMaxBufferSizeMB = other._luceneMaxBufferSizeMB;
       _luceneAnalyzerClass = other._luceneAnalyzerClass;
+      _luceneAnalyzerClassArgs = other._luceneAnalyzerClassArgs;
+      _luceneAnalyzerClassArgTypes = other._luceneAnalyzerClassArgTypes;
+      _luceneQueryParserClass = other._luceneQueryParserClass;
       _enablePrefixSuffixMatchingInPhraseQueries = other._enablePrefixSuffixMatchingInPhraseQueries;
     }
 
     public TextIndexConfig build() {
       return new TextIndexConfig(false, _fstType, _rawValueForTextIndex, _enableQueryCache, _useANDForMultiTermQueries,
           _stopWordsInclude, _stopWordsExclude, _luceneUseCompoundFile, _luceneMaxBufferSizeMB, _luceneAnalyzerClass,
-          _enablePrefixSuffixMatchingInPhraseQueries);
+          CsvParser.serialize(_luceneAnalyzerClassArgs, false, false),
+          CsvParser.serialize(_luceneAnalyzerClassArgTypes, false, false),
```

Comment:
```suggestion
          CsvParser.serialize(_luceneAnalyzerClassArgs, true, false),
          CsvParser.serialize(_luceneAnalyzerClassArgTypes, true, false),
```

Reply:
Changed only _luceneAnalyzerClassArgs to escape comma as the args type should not contain escaped comma at all.